packages/groqd/src/select.ts

```diff
@@ -6,9 +6,11 @@ import {
 } from "./selectionUtils";
 import { extendsBaseQuery, isQuerySchemaTuple } from "./typeGuards";
 import type { FromSelection, Selection } from "./types";
+import { nullToUndefined } from "./nullToUndefined";
 
 export const select = <Conditions extends ConditionRecord>(
-  conditionalSelections: Conditions
+  conditionalSelections: Conditions,
+  shouldCoerceNullValues = false
```

Comment:
might want to eventually move this to being an options arg if we wanna support something like a `keyOn` option, but this works fine in the interim since it's pretty much only internal

Reply:
something to think about here tho is doing something like this:

```ts
const query = q('*').filter().grab({
  foo: q.select({
    '_type == "Type1"': {
      bar: q.string()
    }
  }, true)
})
```
Even tho `shouldCoerceNullValues` is `true` here, the type from this query is:
```
{ foo: ({ bar: string} | null) }[]
```